src/shared/contexts/pinneditems.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {
+  createContext,
+  FC,
+  useEffect,
+  useState,
+  memo,
+  useCallback,
+  ReactElement,
+} from 'react'
+import {CLOUD} from '../constants'
+
+let getPinned
+let putPinned
+let deletePinned
+let postPinned
+
+if (CLOUD) {
+  const pinnedItemFunctions = require('src/client/pinnedItemRoutes')
+  getPinned = pinnedItemFunctions.getPinned
+  putPinned = pinnedItemFunctions.putPinned
+  postPinned = pinnedItemFunctions.postPinned
+  deletePinned = pinnedItemFunctions.deletePinned
+}
+
+export interface PinnedItem {
+  orgID: string
+  userID: string
+  id: string
+  createdAt: string
+  updatedAt: string
+  metadata: {
+    [key: string]: string
+  }
+  type: string
+}
+
+export enum PinnedItemTypes {
+  Dashboard = 'dashboard',
+  Cell = 'cell',
+  Notebook = 'notebook',
+  Task = 'task',
+  Alert = 'alert',
+  UploadCSV = 'uploadCsv',
+  UploadLP = 'uploadLP',
+  ClientLibrary = 'clientLibrary',
+}
+export const PinnedItemsContext = createContext(null)
+
+export const getPinnedItems = async () => {
+  const pinnedRes = await getPinned({})
+  if (pinnedRes.status !== 200) {
+    throw new Error('Unable to retrieve pinned items')
+  } else {
+    return pinnedRes.data
+  }
+}
+
+export const addPinnedItem = async (item: Partial<PinnedItem>) => {
+  const pinnedRes = await postPinned({data: item})
+  if (pinnedRes.status !== 200) {
+    throw new Error('Unable to retrieve pinned items')
+  }
+}
+
+export const deletePinnedItemByParam = async (param: string) => {
+  try {
+    const items = await getPinnedItems()
+    const toDeleteItem = items.find(item =>
+      Object.values(item.metadata).includes(param)
+    )
```

Comment:
so confused, shouldn't this be handled by the API?

Reply:
this is helping us determine based on a parameter whether the request is valid without having to do a separate lookup for the resources using database calls. I just chose to imlpement it on the frontend so we dont make a call to delete in the frist place if we have nothing to delete

Reply:
Totally, I guess my concern is that we end up firing two requests if we want to delete or update, and only 1 if it's invalid ... best case scenario right? Seems like an API concern and not a UI concern

Reply:
Yeah, i could put up a ticket to move this over to the backend, good idea. Backend changes require significant....process